fees/get-protocol.ts

```diff
@@ -1,86 +1,83 @@
-    const block = (await getBlock(timestamp, CHAIN.POLYGON, {}));
-    const graphQueryFees = gql`
+    const beginningOfDay = getTimestampAtStartOfDayUTC(timestamp);
+    const dateId = Math.floor(beginningOfDay / 86400);
+
+    const revenueQuery = gql`
       {
         protocolDay(id: ${dateId}) {
           reservedFuel
           reservedFuelProtocol
+          treasuryRevenue
+          holdersRevenue
         }
       }
     `;
 
-    const graphQueryFeesAllTime = gql`
-      {
-        protocol(id: "1", block: { number: ${block} }) {
-          reservedFuel
-          reservedFuelProtocol
-        }
-      }
-    `;
-    const graphQueryGutsFees = gql`
-      {
-        integratorDay(id: "${dateIdIntegratorDay}") {
-          reservedFuel
-        }
-      }
-    `;
-    const graphQueryGETPrice = gql`
+    const feesQuery = gql`
       {
-        priceOracle(id: "1", block: { number: ${block} }) {
-          price
+        stakingRewards(where: { blockTimestamp_gte: ${beginningOfDay}, blockTimestamp_lt: ${timestamp} }) {
+          totalRewards
+          type
         }
       }
     `;
-    const graphRes = await request(graphUrl, graphQueryFees);
-    const graphResAllTime = await request(graphUrl, graphQueryFeesAllTime);
-    const graphGutsFees = await request(graphUrl, graphQueryGutsFees);
-    const graphGETPrice = await request(graphUrl, graphQueryGETPrice);
 
-    //GET Price in USD
-    const getPrice = parseFloat(graphGETPrice.priceOracle.price);
-    //total fees
-    const finalDailyFee =
-      parseFloat(graphRes.protocolDay.reservedFuel) * getPrice;
+    const graphRevenue = await request(PROTOCOL_SUBGRAPH, revenueQuery);
+    const graphPolyFees = await request(TOKEN_SUBGRAPH_POLYGON, feesQuery);
+    const graphEthFees = await request(TOKEN_SUBGRAPH_ETHEREUM, feesQuery);
+    const getUSD = (await getPrices([PRICE_ID], beginningOfDay))[PRICE_ID].price;
 
-    const finalFeeAllTime =
-      parseFloat(graphResAllTime.protocol.reservedFuel) * getPrice;
+    const integratorFees = parseFloat(graphRevenue.protocolDay.reservedFuel) * getUSD;
+    const dailyHoldersRevenue = parseFloat(graphRevenue.protocolDay.holdersRevenue) * getUSD;
+    const dailyProtocolRevenue = parseFloat(graphRevenue.protocolDay.treasuryRevenue) * getUSD;
+    const dailyRevenue = dailyProtocolRevenue + dailyHoldersRevenue;
 
-    //GUTS fees
-    const gutsFeesDaily =
-      parseFloat(graphGutsFees.integratorDay.reservedFuel) * getPrice;
+    // Transform staking rewards from both Polygon and Ethereum networks into an object indexed by the reward type.
+    // The value of the each type will be the USD amount of GET rewarded using the price at that point in time.
+    const stakingFees = graphEthFees.stakingRewards.concat(graphPolyFees.stakingRewards).reduce(
+      (tally: any, reward: any) => ({
+        [reward.type]: ((tally[reward.type] || 0) + BigInt(reward.totalRewards) / BigInt(10e18)) * getUSD,
```

Comment:
Here, when I ran the code, I have gotten this error:

![image](https://github.com/GETProtocol/dimension-adapters/assets/111286837/4041c11d-4bac-4f32-9d91-8375cd09d9ef)

As a solution, I propose this:
```suggestion
        [reward.type]: ((tally[reward.type] || 0) + Number(BigInt(reward.totalRewards) / BigInt(10e18))) * getUSD,
```

Reply:
Good catch, commiting this change.